code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -0,0 +1,400 @@
+	..()
+
+/mob/living/simple_animal/bot/honkbot/proc/bike_horn() //use bike_horn
+	if (!emagged)
+		if(ckey == null) //check if a player is controlling
+			playsound(loc, honksound, 50, 1, -1)
+		else
+			if (!spam_flag)
+				playsound(loc, honksound, 50, 1, -1)
+				spam_flag = 1 //prevent spam
+		icon_state = "honkbot-c"
+		spawn(5)
+			icon_state = "honkbot[on]"
+		spawn(cooldowntimehorn)
+			spam_flag = 0
+
+	else //emagged honkbots will spam short and memorable sounds.
+
+		if (ckey == null)
+			playsound(loc, pick('sound/items/bikehorn.ogg', 'sound/items/AirHorn2.ogg', 'sound/misc/sadtrombone.ogg', 'sound/items/AirHorn.ogg', 'sound/effects/reee.ogg', 'sound/effects/adminhelp.ogg', 'sound/items/WEEOO1.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bcreep.ogg','sound/magic/Fireball.ogg' ,'sound/effects/pray.ogg', 'sound/voice/hiss1.ogg','sound/machines/buzz-sigh.ogg', 'sound/machines/ping.ogg', 'sound/weapons/flashbang.ogg', 'sound/weapons/bladeslice.ogg'), 50, 0)
+			// to be put at 100 volume: (Weeoo1, bcreep, blaw, hiss, flashbang )
+		else
+			if (!spam_flag)
+				playsound(loc, pick('sound/items/bikehorn.ogg', 'sound/items/AirHorn2.ogg', 'sound/misc/sadtrombone.ogg', 'sound/items/AirHorn.ogg', 'sound/effects/reee.ogg', 'sound/effects/adminhelp.ogg', 'sound/items/WEEOO1.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bcreep.ogg','sound/magic/Fireball.ogg' ,'sound/effects/pray.ogg', 'sound/voice/hiss1.ogg','sound/machines/buzz-sigh.ogg', 'sound/machines/ping.ogg', 'sound/weapons/flashbang.ogg', 'sound/weapons/bladeslice.ogg'), 50, 0)
+				spam_flag = 1 // prevent spam
+		icon_state = "honkbot-e"
+		spawn(30) // keep flashing
+			icon_state = "honkbot[on]"
+		spawn(cooldowntimehorn)
+			spam_flag = 0
+
+/mob/living/simple_animal/bot/honkbot/proc/honk_attack(mob/living/carbon/C) // horn attack
+	if (ckey == null) //check if a player is controlling
+		playsound(loc, honksound, 50, 1, -1)
+	else
+		if(spam_flag == 0)
+			playsound(loc, honksound, 50, 1, -1)
+			spam_flag = 1 // prevent spam
+	icon_state = "honkbot-c"
+	spawn(5)
+		icon_state = "honkbot[on]"
+	spawn(cooldowntimehorn)
+		spam_flag = 0
+
+/mob/living/simple_animal/bot/honkbot/proc/stun_attack(mob/living/carbon/C) // airhorn stun
+	if (ckey == null)
+		playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HOOOOOOOOOOOOONK!!
+	else
+		if(spam_flag == 0)
+			playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HEEEEEEEEEEEENK!!
+	icon_state = "honkbot-c"
+	spawn(5)
+		icon_state = "honkbot[on]"
+	if(spam_flag == 0)
+		if(ishuman(C))
+			C.stuttering = 20
+			C.Jitter(50)
+			C.Knockdown(80)
+			var/mob/living/carbon/human/H = C
+			if (ckey != null) //don't alter AI behavior
+				spam_flag = 1
+			if (!emagged) //HONK once, then leave
+				threatlevel = H.assess_threat(src)
+				threatlevel -= 6
+				//target = old_target
+			else // you really don't want to hit an emagged honkbot
+				threatlevel = H.assess_threat(src)
+				threatlevel = 6 // will never let you go
```

Comment:
This line makes the above line redundant.  There's no point in assessing the threat level if you're just gonna set it to 6 afterwards anyway.